fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -773,8 +773,10 @@ private void handleChargePaidTransaction(final LoanCharge charge, final LoanTran
         LocalDate startDate = getDisbursementDate();
         List<LoanRepaymentScheduleInstallment> installments = getRepaymentScheduleInstallments();
         for (final LoanRepaymentScheduleInstallment installment : installments) {
-            if (installmentNumber == null && charge.isDueForCollectionFromAndUpToAndIncluding(startDate, installment.getDueDate())) {
-
+            boolean isDue = installment.isFirstPeriod()
+                    ? charge.isDueForCollectionFromIncludingAndUpToAndIncluding(startDate, installment.getDueDate())
+                    : charge.isDueForCollectionFromAndUpToAndIncluding(startDate, installment.getDueDate());
```

Comment:
For this condition, do we need to check from `disbursementDate` of loan?
A charge is due if it is less than `installment.getDueDate()`

Reply:
Not entirely true. We are going through on each installment and the startDate need to be included only for the very first period. For 2nd period, the start date is not inclusive as the start date of the second installment and due date of the 1st installment is the same